integrations/class-base.php

```diff
@@ -128,13 +163,11 @@ public function show_select_or_input() {
 
 			<?php else : // select menu ?>
 
-				<select id="<?php echo $this->context;?>-affiliate" name="<?php echo $this->context;?>_affiliate" class="<?php echo $this->context;?>-select">
+				<?php $affiliates = $this->get_affiliates_select_list( $affiliate_list ); ?>
 
-				<option value="0"><?php _e( 'Select', 'affiliatewp-checkout-referrals' ); ?></option>
-				<?php foreach ( $affiliate_list as $affiliate_id => $user_id ) :
-					$user_info = get_userdata( $user_id );
-				?>
-					<option value="<?php echo $affiliate_id; ?>"><?php echo $user_info->$display; ?></option>
+				<select id="<?php echo $this->context;?>-affiliate" name="<?php echo $this->context;?>_affiliate" class="<?php echo $this->context;?>-select">
+				<?php foreach ( $affiliates as $affiliate_id => $affiliate_display ) : ?>
+					<option value="<?php echo $affiliate_id; ?>"><?php echo $affiliate_display; ?></option>
```

Comment:
Need `esc_attr()` on the `$affiliate_id` here.
```suggestion
					<option value="<?php echo esc_attr( $affiliate_id ); ?>"><?php echo $affiliate_display; ?></option>
```